### PR TITLE
[Debezium] Adding all Timestamp converters

### PR DESCRIPTION
**Pull request by Tang8330**
## Changes

Refactoring the following converters:
* Timestamp
* KafkaConnectTimestamp
* MicroTimestamp
* NanoTimestamp

### CODE REVIEW

File: lib/debezium/types.go [hunk 1]
```diff
@@ -33,15 +33,15 @@ const (
...
-	DateTimeKafkaConnect SupportedDebeziumType = "org.apache.kafka.connect.data.Timestamp"
+	MicroTimestamp        SupportedDebeziumType = "io.debezium.time.MicroTimestamp"
+	NanoTimestamp         SupportedDebeziumType = "io.debezium.time.NanoTimestamp"
+	Date                  SupportedDebeziumType = "io.debezium.time.Date"
+	Year                  SupportedDebeziumType = "io.debezium.time.Year"
+	DateTimeWithTimezone  SupportedDebeziumType = "io.debezium.time.ZonedTimestamp"
+	MicroDuration         SupportedDebeziumType = "io.debezium.time.MicroDuration"
+	DateKafkaConnect      SupportedDebeziumType = "org.apache.kafka.connect.data.Date"
+	Timestamp             SupportedDebeziumType = "io.debezium.time.Timestamp"
+	TimestampKafkaConnect SupportedDebeziumType = "org.apache.kafka.connect.data.Timestamp"
```
Comment (Tang8330) [inline lib/debezium/types.go:44]: Only change here is that I renamed this to `TimestampKafkaConnect` to be more correct

File: lib/debezium/types.go [hunk 2]
```diff
@@ -33,23 +33,26 @@ const (
 	EnumSet SupportedDebeziumType = "io.debezium.data.EnumSet"
 	UUID    SupportedDebeziumType = "io.debezium.data.Uuid"
 
-	Timestamp            SupportedDebeziumType = "io.debezium.time.Timestamp"
-	MicroTimestamp       SupportedDebeziumType = "io.debezium.time.MicroTimestamp"
-	NanoTimestamp        SupportedDebeziumType = "io.debezium.time.NanoTimestamp"
+	// Dates
```
Comment (Tang8330) [inline lib/debezium/types.go:36]: Just organized this

File: lib/debezium/schema_test.go [hunk 3]
```diff
@@ -85,11 +85,6 @@ func TestField_ToKindDetails(t *testing.T) {
 		assert.Equal(t, typing.Integer, Field{Type: Int32}.ToKindDetails())
 		assert.Equal(t, typing.Integer, Field{Type: Int64}.ToKindDetails())
 	}
-	{
```
Comment (nathan-artie) [inline lib/debezium/schema_test.go:88]: Is this change intentional?

Comment (Tang8330) [reply lib/debezium/schema_test.go:88]: Deleting this? No, must've happened from the merge. Thanks for the catch

File: lib/typing/ext/variables.go [hunk 4]
```diff
@@ -35,3 +35,10 @@ var SupportedTimeFormatsLegacy = []string{
 	PostgresTimeFormatNoTZ,
 	AdditionalTimeFormat,
 }
+
+// RFC 339 variants
```
Comment (nathan-artie) [inline lib/typing/ext/variables.go:39]: Nit: missing a `3` in `RFC 3339`

Comment (Tang8330) [reply lib/typing/ext/variables.go]: Thanks!

File: lib/debezium/converters/timestamp.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+func (Timestamp) Convert(value any) (any, error) {
+	castedValue, err := typing.AssertType[int64](value)
+	if err != nil {
+		return nil, err
+	}
+
+	// Represents the number of milliseconds since the epoch, and does not include timezone information.
+	return ext.NewExtendedTime(time.UnixMilli(castedValue).In(time.UTC), ext.DateTimeKindType, ext.RFC3339Millisecond), nil
```
Comment (nathan-artie) [inline lib/debezium/converters/timestamp.go:23]: Do we need to call `IsValid()` on the return value from `NewExtendedTime`?

File: lib/debezium/converters/timestamp_test.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	}
+	{
+		// Valid conversion
+		converted, err := MicroTimestamp{}.Convert(int64(1_712_609_795_827_923))
+		assert.NoError(t, err)
+		assert.Equal(t, "2024-04-08T20:56:35.827923Z", converted.(*ext.ExtendedTime).String(""))
+	}
+	{
+		// micros is preserved despite it being all zeroes.
+		converted, err := MicroTimestamp{}.Convert(int64(1_712_609_795_827_123))
```
Comment (nathan-artie) [inline lib/debezium/converters/timestamp_test.go:47]: Should the input for this test end in `_827_123`?

File: lib/debezium/converters/timestamp.go (hunk 5 above)
Comment (Tang8330) [reply lib/debezium/converters/timestamp.go:23]: I don't think so, if we are able to correctly get `time.Time` and parse the year to be larger than 9999, it still passes `time.Format(...)`

https://go.dev/play/p/tOJR2Ho6EZi

File: lib/debezium/converters/timestamp_test.go (hunk 6 above)
Comment (Tang8330) [reply lib/debezium/converters/timestamp_test.go:47]: Yes it should. Updating the test

**Review by nathan-artie — APPROVED** [review]
